### PR TITLE
21701-Add-primitive-to-change-the-window-icon

### PR DESCRIPTION
**Pull request by VincentBlondeau**
add primitive for changing the icon of the main window + TESTS
use basename instead of name on file references

### CODE REVIEW

**Comment by VincentBlondeau** [comment]
https://pharo.fogbugz.com/f/cases/21701/Add-primitive-to-change-the-window-icon

**Comment by VincentBlondeau** [comment]
The latest VM now integrates the primitive. But I do not know if it used in the CI.

**Comment by jecisc** [comment]
There is now a conflict

src/Files/File.class.st

**Comment by VincentBlondeau** [comment]
@jecisc No conflict anymore

**Comment by VincentBlondeau** [comment]
I do not understand why the tests are failing ... they seem to be unrelated

File: src/Graphics-Tests/DisplayScreenTest.class.st [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
...85 239 55 34 228 83 130 231 196 11 59 142 52 92 108 212 97 53 206 23 152 241 227 17 233 57 79 232 204 26 140 185 250 53 184 70 200 184 130 103 160 18 193 116 213 82 168 90 202 97 220 107 192 42 180 164 226 68 162 150 84 60 23 241 162 15 160 209 2 64 21 24 2 249 30 174 254 5 118 122 132 39 126 119 69 156 18 43 87 63 246 230 54 184 237 200 9 138 217 22 172 115 30 200 160 201 20 225 213 68 137 59 234 188 57 92 43 171 85 165 197 139 189 240 222 2 132 128 2 23 166 67 206 215 218 113 45 5 71 98 213 124 75 219 187 61 250 70 226 88 126 71 198 48 227 199 77 59 79 76 4 4 181 53 88 9 149 129 153 144 191 122 187 4 2 155 242 134 174 156 123 149 238 66 241 201 166 28 184 125 75 10 63 138 163 23 67 197 136 82 160 76 87 149 111 158 40 147 118 226 8 66 119 54 51 231 51 105 18 20 56 42 232 75 88 6 245 239 239 170 242 200 202 124 209 202 105 0 175 254 6 220 247 105 69 103 12 208 34 176 131 184 101 88 194 42 19 105 60 94 124 160 86 199 223 146 138 71 114 93 171 87 190 85 229 187 231 170 188 52 232 115 182 16 46 90 8 24 129 124 74 248 212 141 121 246 119 123 151 45 91 22 129 180 11 125 45 14 158 43 132 54 214 62 102 131 3 179 149 132 83 21 203 120 217 178 183 211 37 227 114 209 134 191 194 185 25 226 1
```
Comment (jecisc) [inline src/Graphics-Tests/DisplayScreenTest.class.st]: @VincentBlondeau You need to call "super setUp" here to get the tests to pass.

Comment (jecisc) [inline src/Graphics-Tests/DisplayScreenTest.class.st]: For tests maybe we can find a smaller icon?

**Review by jecisc — CHANGES_REQUESTED** [review]

**Comment by philippeback** [comment]
Any chance this can be working?

**Comment by jecisc** [comment]
@philippeback Once the vm will include the primitive I think it should work. Else I think Vincent would not have done the PR yet.

**Comment by VincentBlondeau** [comment]
@philippeback The primitive is integrated in the latest VM, you should be able to use it

**Comment by MarcusDenker** [comment]
ok to merge?

**Comment by jecisc** [comment]
@MarcusDenker I tested the generated image and it works with the current stable VM.

**Review by jecisc — APPROVED** [review]

**Pull request merged**